website_blog_layout_options/models/blog_post.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    This module copyright (C) 2015 Therp BV (<http://therp.nl>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields, api
+from lxml import etree
+
+
+class BlogPost(models.Model):
+
+    _inherit = 'blog.post'
+
+    @api.one
+    @api.onchange('extract_auto', 'display_type', 'content', 'teaser_input')
+    def _extract_teaser(self):
+        if self.display_type == "teaser":
+            # no empty teasers
+            if (self.teaser_input or '') and (not self.extract_auto):
+                self.teaser = self.teaser_input
+            else:
+                res = ""
+                # limit length to roughly 3-4 lines.
+                teaser_length = 500
+                parser = etree.HTMLParser()
+                if self.content:
```

Comment:
consider using https://github.com/OCA/OCB/blob/8.0/openerp/tools/mail.py#L143 to avoid some of the more common pitfalls